config.c

```diff
@@ -25,6 +33,66 @@ void finish_config(struct mako_config *config) {
 	free(config->format);
 }
 
+/* Parse between 1 and 4 integers, comma separated, from the provided string.
+ * Depending on the number of integers provided, the four fields of the `out`
+ * struct will be initialized following the same rules as the CSS "margin"
+ * property. Returns 0 on success, -1 on parsing failure. */
+int parse_directional(const char *directional_string,
+	struct directional *out) {
+	int error = 0;
+	char *components = strdup(directional_string);
+
+	char *token, *saveptr, *endptr = NULL;
+	int32_t values[] = {0, 0, 0, 0};
+	int count = 0;
+	int number;
+
+	token = strtok_r(components, ",", &saveptr);
+	for (count = 0; count < 4; count++) {
+	if (token == NULL) {
+	break;
+	}
+
+	errno = 0;
+
+	number = strtol(token, &endptr, 10);
+	if (errno || endptr == token) {
+	// There were no digits, or something else went horribly wrong.
+	error = -1;
+	break;
+	}
+
+	values[count] = number;
+	token = strtok_r(NULL, ",", &saveptr);
+	}
+
+	if (error == 0) {
+	switch (count) {
+	case 1: // All values are the same
+	out->top = out->right = out->bottom = out->left = values[0];
+	break;
+	case 2: // Vertical, horizontal
+	out->top = out->bottom = values[0];
+	out->right = out->left = values[1];
+	break;
+	case 3: // Top, horizontal, bottom
+	out->top = values[0];
+	out->right = out->left = values[1];
+	out->bottom = values[2];
+	break;
+	case 4: // Top, right, bottom, left
+	out->top = values[0];
+	out->right = values[1];
+	out->bottom = values[2];
+	out->left = values[3];
+	break;
+	}
+	}
+
+	free(components);
+	return error;
+}
```

Comment:
You need to add some indents to this function my friend

Reply:
Agh, my spaces to tabs function failed me horribly!